pkg/sideinputs/synchronizer/synchronizer.go

```diff
@@ -104,13 +106,17 @@ func startSideInputSynchronizer(ctx context.Context, watch kvs.KVWatcher, mountP
 				log.Warnw("nil value received from Side Input watcher")
 				continue
 			}
-			log.Debug("Side Input value received ",
+			log.Infow("Side Input value received ",
 				zap.String("key", value.Key()), zap.String("value", string(value.Value())))
 			p := path.Join(mountPath, value.Key())
-			// Write changes to disk
-			err := utils.UpdateSideInputFile(ctx, p, value.Value())
-			if err != nil {
-				log.Errorw("Failed to update Side Input value %s", zap.Error(err))
+			// Check if the value has changed from the last update
+			if prevValue == nil || bytes.Equal(prevValue, value.Value()) == false {
```

Comment:
Do the check here means when a pod starts, there will be a consecutive file changes - one is done by the init container, the other one is done by this. Shall we do it in the function of `UpdateSideInputFile()`? To read the old file content and compare?

Reply:
Done